File: app/src/main/java/br/org/cesar/discordtime/stickysessions/data/remote/model/SessionRemote.java

package br.org.cesar.discordtime.stickysessions.data.remote.model;

import com.google.firebase.database.ServerValue;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.google.firebase.database.Exclude;
import com.google.firebase.database.IgnoreExtraProperties;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@IgnoreExtraProperties
public class SessionRemote {

    @Exclude
    public String id;
    @SerializedName("topics")
    @Expose
    public List<String> topics = null;
    @SerializedName("date")
    @Expose
    public long date;

    public SessionRemote() {}

    public SessionRemote(String id, List<String> topics) {
        this.id = id;
        this.topics = topics;
    }

    @Exclude
    public Map<String, Object> toMap() {
        HashMap<String, Object> result = new HashMap<>();
        result.put("topics", topics);
        result.put("date", ServerValue.TIMESTAMP);
        return result;
    }

}

File: app/src/main/java/br/org/cesar/discordtime/stickysessions/data/remote/model/SessionRemote.kt
package br.org.cesar.discordtime.stickysessions.data.remote.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class SessionRemote(
        @Expose @SerializedName("id") var id: String,
        @Expose @SerializedName("topics") var topics: List<String>,
        @Expose @SerializedName("timestamp") var timestamp: Long,
        @Expose @SerializedName("title") var title: String?,
        @Expose @SerializedName("description") var description: String?,
        @Expose @SerializedName("color") var color: String?
) {
    constructor(id: String, topics: List<String>, title: String, description: String?,
                color: String?) : this(id,topics,0,title,description,color)

    constructor(fromFirebase: SessionRemoteFirebase):
            this(fromFirebase.id, fromFirebase.topics,
                    1000L * (fromFirebase.timestamp?.seconds ?: 0), fromFirebase.title,
                    fromFirebase.description, fromFirebase.color)
}

